h5netcdf/core.py

```diff
@@ -724,6 +748,23 @@ def _create_child_variable(
         else:
             self._root._check_valid_netcdf_dtype(dtype)
 
+        # compound type handling
+        if np.dtype(dtype).kind == "c":
+            if not isinstance(dtype, CompoundType):
+                # todo check compound types for
+                #  existing complex types which may be used
+                itemsize = np.dtype(dtype).itemsize
+                try:
+                    width = {8: "FLOAT", 16: "DOUBLE"}[itemsize]
+                except KeyError:
+                    raise TypeError(
+                        "Currently only 'complex64' and 'complex128' dtypes are allowed."
+                    )
```

Comment:
This results in a slightly better traceback ("direct cause of" rather than "during the handling of")
```suggestion
                except KeyError as e:
                    raise TypeError(
                        "Currently only 'complex64' and 'complex128' dtypes are allowed."
                    ) from e
```